### PR TITLE
Fix node-gyp build error on MacBook 2017

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

Fix node-gyp build error on MacBook 2017

```
error /Users/tuyennguyen/Projects/lodestar/node_modules/snappy: Command failed.
Exit code: 1
Command: prebuild-install || node-gyp rebuild
Arguments: 
Directory: /Users/tuyennguyen/Projects/lodestar/node_modules/snappy
Output:
prebuild-install WARN install No prebuilt binaries found (target=16.13.1 runtime=node arch=x64 libc= platform=darwin)
```

**Description**

Add latest `node-gyp` dev dependency

@ChainSafe/lodestar  could you help try this branch in your environment? Thanks 🙏 

- [ ] @philknows
- [ ] @wemeetagain 
- [x] @dapplion 
- [ ] @dadepo 
- [x] @g11tech

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -512,7 +512,78 @@
     protobufjs "^6.11.2"
     uint8arrays "^3.0.0"
 
-"@chainsafe/persistent-merkle-tree@^0.4.2":
+"@chainsafe/lodestar-api@^0.37.0":
```
Comment (wemeetagain) [inline yarn.lock]: extra dependencies

Comment (wemeetagain) [reply yarn.lock]: See #4186

**Review by dapplion — DISMISSED** [review]
Did
```
rm -r node_modules
yarn
```
and all packages including snappy build fine :+1:

**Comment by g11tech** [comment]
> Did
> 
> ```
> rm -r node_modules
> yarn
> ```
> 
> and all packages including snappy build fine +1

ditto! :checkered_flag: 

But if I remove yarn.lock (+ rm -rf node_modules) and install/build again: this happens: might not be related to this
```typescript
bash-5.1# yarn build
yarn run v1.22.15
$ yarn clean && lerna run build
$ rm -rf ./packages/*/lib ./packages/*/*.tsbuildinfo
lerna notice cli v4.0.0
lerna info Executing command in 15 packages: "yarn run build"
@chainsafe/lodestar-params: $ tsc -p tsconfig.build.json
@chainsafe/lodestar-utils: $ tsc -p tsconfig.build.json
@chainsafe/lodestar-params: ../../node_modules/@types/tar/index.d.ts:239:36 - error TS2709: Cannot use namespace 'MiniPass' as a type.
@chainsafe/lodestar-params: 239 export interface ReadEntry extends MiniPass, HeaderProperties {
@chainsafe/lodestar-params:                                        ~~~~~~~~
@chainsafe/lodestar-params: Found 1 error in ../../node_modules/@types/tar/index.d.ts:239
@chainsafe/lodestar-params: error Command failed with exit code 2.
@chainsafe/lodestar-params: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run build exited 2 in '@chainsafe/lodestar-params'
lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by twoeths** [comment]
e2e passed @dapplion

**Review by dapplion — APPROVED** [review]

**Pull request merged**